lib/loader/loader.js

```diff
@@ -217,6 +217,11 @@ var loader, define, requireModule, require, requirejs;
   };
 
   define = function(name, deps, callback) {
+    var module = registry[name];
+    if (module && module.state !== 'new' && module.state !== 'errored') {
```

Comment:
I agree, making it conditional increases the likelihood of modules from one package intermixing versions

engine1.js   defines   a/b  a/c  a/d

engine2.js   defines   a/b  a/c   a/d

if a/b is erroed, and a/c and a/d are finalized,  you with try a/b from the other engine.